cast/src/lib.rs

```diff
@@ -490,6 +505,46 @@ impl SimpleCast {
             Ok(value)
         }
     }
+    /// Decodes abi-encoded hex input and output
+    ///
+    /// ```
+    /// use cast::SimpleCast as Cast;
+    ///
+    /// fn main() -> eyre::Result<()> {
+    ///     let mut test_string =
+    ///     String::from("0xf242432a0000000000000000000000008dbd1b711dc621e1404633da156fcc779e1c6f3e000000000000000000000000d9f3c9cc99548bf3b44a43e0a2d07399eb918adc000000000000000000000000000000000000000000000000000000000000002a000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000000");
+    ///     let mut signature = String::from("function safeTransferFrom(address, address, uint256, uint256, bytes)");
+    ///     let (mut input, mut output) = Cast::abi_decode(signature, Some(test_string), None);
+    ///     assert_eq!("[Address(0x8dbd1b711dc621e1404633da156fcc779e1c6f3e), Address(0xd9f3c9cc99548bf3b44a43e0a2d07399eb918adc), Uint(42), Uint(1), Bytes([])]",
+    ///     format!("{:?}", input.unwrap()));
+    ///     test_string = String::from("0x0000000000000000000000000000000000000000000000000000000000000001");
+    ///     signature = String::from("function balanceOf(address,uint256) view returns(uint256)");
+    ///     let (input2, output2) = Cast::abi_decode(signature, None, Some(test_string));
+    ///     assert_eq!("[Uint(1)]",
+    ///     format!("{:?}", output2.unwrap()));
+    ///     Ok(())
+    /// }    
+    /// ```
+    pub fn abi_decode(
+        sig: String,
+        calldata: Option<String>,
+        output: Option<String>,
+    ) -> (Result<Vec<Token>>, Result<Vec<Token>>) {
+        let func = foundry_utils::IntoFunction::into(sig);
+        let mut decoded_input = Ok(vec![]);
+        let mut decoded_output = Ok(vec![]);
+        if let Some(mut x) = calldata {
```

Comment:
It doesn't make much sense to pass both inputs and outputs to the same call I think? IMO only take 1 hex string and add a --output bool flag which will trigger decoding using the output instead of the input (vs doing both)